jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/RuntimeDataServiceImpl.java

```diff
@@ -799,15 +801,98 @@ public UserTaskInstanceDesc getTaskById(Long taskId) {
 		applyQueryFilter(params, filter);
 		return (List<TaskSummary>) commandService.execute(new QueryNameCommand<List<TaskSummary>>("TasksByStatusByProcessId", params));
 	}
+
+
+    @Override
+    public List<TaskSummary> getTasksAssignedAsBusinessAdministratorByStatus(String userId,
+                                                                             List<Status> statuses,
+                                                                             QueryFilter filter) {
+        return ((InternalTaskService)taskService).getTasksAssignedAsBusinessAdministratorByStatus(userId, filter.getLanguage(), statuses);
+    }
+
+    @Override
+    public TaskSummaryQueryBuilder taskSummaryQuery(String userId) {
+        return new TaskSummaryQueryBuilderImpl(userId, taskService);
+    }
+
+    @Override
+    public List<TaskSummary> getTasksByVariable( String userId, String variableName, Status... statuses ) {
+        TaskSummaryQueryBuilder queryBuilder = new TaskSummaryQueryBuilderImpl(userId, taskService).intersect();
+        addVariableNameAndVariableValueParameters(queryBuilder, variableName);
+
+        if( statuses != null && statuses.length > 0 ) {
+            queryBuilder.status(statuses);
+        }
+
+        return queryBuilder.build().getResultList();
+    }
+
+    @Override
+    public List<TaskSummary> getTasksByVariable( String userId, String variableName, List<Status> statuses,
+            int offset, int total ) {
```

Comment:
can we use QueryContext instead of int for offset and total? I agree that QueryFilter should not be used but QueryContext is more than paging, it has the sorting in it as well